code/modules/fish/fish_tank.dm

```diff
@@ -350,23 +351,18 @@
 	var/fish = pick(fish_list)						//Select a fish from the fish in the tank
 	if(prob(25))									//25% chance to be a dud (blank) egg
 		fish = "dud"
-	var/obj/item/fish_eggs/egg_path	= null			//Create empty variable to receive the egg_path
-	egg_path = fish_eggs_list[fish]					//Locate the corresponding path from fish_eggs_list that matches the fish
-	if(!egg_path)									//The fish wasn't located in the fish_eggs_list, potentially due to a typo, so return a dud egg
-		return /obj/item/fish_eggs
-	else											//The fish was located in the fish_eggs_list, so return the proper egg
-		return egg_path
+	var/obj/item/fish_eggs/egg_path	= fish_eggs_list[fish]					//Locate the corresponding path from fish_eggs_list that matches the fish
+	return egg_path									//The fish was located in the fish_eggs_list, so return the proper egg
 
 /obj/machinery/fishtank/proc/harvest_eggs(var/mob/user)
 	if(!egg_list.len)									//Can't harvest non-existant eggs
 		return
 
-	for(var/i in egg_list.len)						//Loop until you've harvested all the eggs
-		var/obj/item/fish_eggs/egg = pick(egg_list)	//Select an egg at random
-		egg = new egg(get_turf(user))				//Spawn the egg at the user's feet
-		egg_list.Remove(egg)						//Remove the egg from the egg_list
+	for(var/i = 1 to egg_list.len)						//Loop until you've harvested all the eggs
+		var/obj/item/fish_eggs/egg = egg_list[i]	//Go through the eggs
+		new egg(get_turf(user))						//Spawn the egg at the user's feet
 
-	egg_list.Cut()									//Destroy any excess eggs, clearing the egg_list
+	egg_list = list()								//Destroy any excess eggs, clearing the egg_list
```

Comment:
why this?

Reply:
I wanted to explicitly clean the list, but Cut() does the same thing so yeah
If Cut() is better, then I'll change it back

Reply:
![image](https://user-images.githubusercontent.com/6307265/35192422-23ebabca-fe92-11e7-8ec2-8fef44abee8a.png)

Reply:
list.len = 0 too
they're all basically the same